core/src/main/scala/kafka/admin/MetadataQuorumCommand.scala

```diff
@@ -0,0 +1,186 @@
+        "A short summary of the quorum status and the other provides detailed information about the status of replication.")
+      .action(storeTrue())
+    val replicationArgs = describeParser.addArgumentGroup("Replication")
+    replicationArgs
+      .addArgument("--replication")
+      .help("Detailed information about the status of replication")
+      .action(storeTrue())
+  }
+
+  private def handleDescribeReplication(admin: Admin): Unit = {
+    val quorumInfo = admin.describeMetadataQuorum().quorumInfo().get()
+    val leaderId = quorumInfo.leaderId()
+    val leader = quorumInfo.voters().asScala.filter(_.replicaId() == leaderId).head
+    // Find proper columns width
+    var (maxReplicaIdLen, maxLogEndOffsetLen, maxLagLen, maxLastFetchTimeMsLen, maxLastCaughtUpTimeMsLen) =
+      (15, 15, 15, 15, 18)
+    (quorumInfo.voters().asScala ++ quorumInfo.observers().asScala).foreach { voter =>
+      maxReplicaIdLen = Math.max(maxReplicaIdLen, voter.replicaId().toString.length)
+      maxLogEndOffsetLen = Math.max(maxLogEndOffsetLen, voter.logEndOffset().toString.length)
+      maxLagLen = Math.max(maxLagLen, (leader.logEndOffset() - voter.logEndOffset()).toString.length)
+      maxLastFetchTimeMsLen = Math.max(maxLastFetchTimeMsLen, leader.lastFetchTimeMs().orElse(-1).toString.length)
+      maxLastCaughtUpTimeMsLen =
+        Math.max(maxLastCaughtUpTimeMsLen, leader.lastCaughtUpTimeMs().orElse(-1).toString.length)
+    }
+    println(
+      s"%${-maxReplicaIdLen}s %${-maxLogEndOffsetLen}s %${-maxLagLen}s %${-maxLastFetchTimeMsLen}s %${-maxLastCaughtUpTimeMsLen}s %-15s "
+        .format("ReplicaId", "LogEndOffset", "Lag", "LastFetchTimeMs", "LastCaughtUpTimeMs", "Status")
+    )
+
+    def printQuorumInfo(infos: Seq[QuorumInfo.ReplicaState], status: String): Unit =
+      infos.foreach { voter =>
+        println(
+          s"%${-maxReplicaIdLen}s %${-maxLogEndOffsetLen}s %${-maxLagLen}s %${-maxLastFetchTimeMsLen}s %${-maxLastCaughtUpTimeMsLen}s %-15s "
+            .format(
+              voter.replicaId(),
+              voter.logEndOffset(),
+              leader.logEndOffset() - voter.logEndOffset(),
+              voter.lastFetchTimeMs().orElse(-1),
+              voter.lastCaughtUpTimeMs().orElse(-1),
+              status
+            ))
+      }
+    printQuorumInfo(Seq(leader), "Leader")
+    printQuorumInfo(quorumInfo.voters().asScala.filter(_.replicaId() != leaderId).toSeq, "Follower")
+    printQuorumInfo(quorumInfo.observers().asScala.toSeq, "Observer")
+  }
+
+  private def handleDescribeStatus(admin: Admin): Unit = {
+    val clusterId = admin.describeCluster().clusterId().get()
+    val quorumInfo = admin.describeMetadataQuorum().quorumInfo().get()
+    val leaderId = quorumInfo.leaderId()
+    val leader = quorumInfo.voters().asScala.filter(_.replicaId() == leaderId).head
+    val maxLagFollower = quorumInfo
+      .voters()
+      .asScala
+      .filter(_.replicaId() != leaderId)
+      .minBy(_.logEndOffset())
```

Comment:
This raises an error if there is only one voter. I think we can just get rid of the `filter`? Then the max follower lag is just 0 if there is only one voter. We need another adjustment for `maxFollowerLagTimeMs` so that it also ends up as 0 when there is only one voter.

Reply:
Good catch, I also added a method `testOnlyOneBrokerAndOneController` for this case.